ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -508,6 +508,10 @@ private void compile(String command, boolean resetTaskIds, boolean deferClose) t
     perfLogger.PerfLogBegin(CLASS_NAME, PerfLogger.COMPILE);
     lDrvState.stateLock.lock();
     try {
+      if (lDrvState.driverState == DriverState.DESTROYED || lDrvState.driverState == DriverState.CLOSED) {
+        LOG.warn("FAILED: Query command could not be compiled because driver has been cancelled, closed or destroyed.");
```

Comment:
Maybe better:
_Query compilation failed as the driver was canceled, closed, or destroyed_

Reply:
But there is no driverState named canceled, i think the closed of driverStates include canceled

Reply:
ok